rom/laghos_rom.cpp

```diff
@@ -192,33 +192,48 @@ void ROM_Sampler::Finalize(const double t, const double dt, Vector const& S, Arr
         }
     }
 
+    if (!writeSnapshots)
+    {
+        if (!useXV)
+        {
+            input.dimX = generator_X->getSpatialBasis()->numColumns();
```

Comment:
Why is this necessary?

Reply:
It took me a while to remember, but I remember now. Normally, the user gives dimX, dimV, and dimE for the online phase. To calculate the residual without the user supplying this information, we need to figure it out from the offline phase, which is why this is required. This updates the romOptions dimensions for the residual.

Reply:
No longer changes the input, but copies the dim to a file since the greedy algorithm is now split across multiple runs of laghos rather than doing the do-while loop.